lxd/db/cluster/certificate_projects.mapper.go

```diff
@@ -71,14 +77,23 @@ func GetCertificateProjects(ctx context.Context, tx *sql.Tx, certificateID int)
 		return nil, fmt.Errorf("Failed to fetch from \"certificates_projects\" table: %w", err)
 	}
 
-	result := make([]Project, len(objects))
-	for i, object := range objects {
-		project, err := GetProjects(ctx, tx, ProjectFilter{ID: &object.ProjectID})
-		if err != nil {
-			return nil, err
+	// We will need at least one filter to apply an ID to, if we have any.
+	if len(filters) == 0 && len(objects) > 0 {
+		filters = []ProjectFilter{{}}
+	}
+
+	// Ensure we have a complete set of filters for each returned ID.
+	idFilters := make([]ProjectFilter, 0, len(objects)*len(filters))
+	for _, object := range objects {
+		for _, filter := range filters {
+			filter.ID = &object.ProjectID
+			idFilters = append(idFilters, filter)
```

Comment:
This approach seems a bit odd to me, I'm going to come back to you on this when I'm on my pc.